ui/src/dashboards/components/QueryEditor.tsx

```diff
@@ -0,0 +1,465 @@
+            onBlur={this.handleBlurEditor}
+            onFocus={this.handleFocusEditor}
+            onKeyDown={this.handleKeyDownEditor}
+            onDblClick={this.handleUnmountEditor}
+            editorDidMount={this.handleMountEditor}
+            onBeforeChange={this.handleUpdateEditorValue}
+          />
+        </div>
+        <div
+          className={classnames('varmoji', {
+            'varmoji-rotated': isTemplating,
+            focus: focused,
+          })}
+        >
+          <div className="varmoji-container">
+            <div className="varmoji-front">
+              <QueryStatus status={status}>
+                <button
+                  onMouseDown={this.handleToggleFocus}
+                  onClick={this.handleToggleTemplateValues}
+                  className={classnames(
+                    'btn btn-xs btn-info query-editor--submit',
+                    {
+                      disabled: isTemplating,
+                    }
+                  )}
+                >
+                  {this.templateToggleStatus}
+                </button>
+                <button
+                  className={classnames(
+                    'btn btn-xs btn-primary query-editor--submit',
+                    {disabled: this.isDisabled}
+                  )}
+                  onClick={this.handleUpdate}
+                >
+                  Submit Query
+                </button>
+              </QueryStatus>
+            </div>
+            <div className="varmoji-back">
+              {isTemplating ? (
+                <TemplateDrawer
+                  onClickTempVar={this.handleClickTempVar}
+                  templates={filteredTemplates}
+                  selected={selectedTemplate}
+                  onMouseOverTempVar={this.handleMouseOverTempVar}
+                  handleClickOutside={this.handleCloseDrawer}
+                />
+              ) : null}
+            </div>
+          </div>
+        </div>
+      </div>
+    )
+  }
+
+  private hideTemplateValues = (): void => {
+    this.setQueryValue(this.state.queryText)
+  }
+
+  private handleToggleFocus = (e: MouseEvent<HTMLButtonElement>): void => {
+    e.stopPropagation()
+    e.preventDefault()
+
+    this.editor.focus()
+  }
+
+  private handleToggleTemplateValues = (): void => {
+    const {isShowingTemplateValues} = this.state
+
+    if (isShowingTemplateValues) {
+      this.hideTemplateValues()
+    } else {
+      this.showTemplateValues()
+    }
+  }
+
+  private get isDisabled(): boolean {
+    return this.state.isTemplating || this.state.isShowingTemplateValues
+  }
+
+  private get queryCodeClassName(): string {
+    const {focused} = this.state
+
+    return classnames('query-editor--code', {focus: focused})
+  }
+
+  private get templateToggleStatus(): string {
+    if (this.state.isShowingTemplateValues) {
+      return HIDE_TEMPLATES
+    }
+
+    return SHOW_TEMPLATES
+  }
+
+  private showTemplateValues = async (): Promise<void> => {
```

Comment:
would you consider placing this method just before or after `hideTemplateValues` for readability purposes?